src/main/scala/viper/gobra/ast/internal/Program.scala

```diff
@@ -346,6 +352,14 @@ case class Access(e: Accessible, p: Expr)(val info: Source.Parser.Info) extends
   require(p.typ.isInstanceOf[PermissionT], s"expected an expression of permission type but got $p.typ")
 }
 
+case class ExprTerminationMeasure(exp:Expr)(val info: Source.Parser.Info) extends Assertion
+case class UnderscoreTerminationMeasure()(val info:Source.Parser.Info) extends Assertion
+case class StarTerminationMeasure()(val info: Source.Parser.Info) extends Assertion
+case class ConditionalMeasure()(val info:Source.Parser.Info) extends  Assertion
+case class ConditionalMeasureExpression( vector: Vector[Expr] ,condition:Expr) (val info:Source.Parser.Info) extends Assertion
+case class ConditionalMeasureUnderscore( condition:Expr) (val info:Source.Parser.Info) extends Assertion
+case class ConditionalMeasureAdditionalStar()(val info:Source.Parser.Info) extends Assertion
+
```

Comment:
I would not make this classes subtypes of `Assertion`. Instead, I would introduce a sealed trait called `DecreaseClause` and change all classes to be a subtype of `DecreaseClause`. For example, I would modify
```scala
case class ExprTerminationMeasure(exp:Expr)(val info: Source.Parser.Info) extends Assertion
```
to
```scala
case class DecreaseExpr(exp:Expr)(val info: Source.Parser.Info) extends DecreaseClause
```

and I would change
```scala
case class UnderscoreTerminationMeasure()(val info:Source.Parser.Info) extends Assertion
```
to
```scala
case class DecreaseUnderscore(val info:Source.Parser.Info) extends DecreaseClause
```